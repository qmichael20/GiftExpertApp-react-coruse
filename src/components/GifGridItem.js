import React from "react";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__pulse" align="center">
      <img src={url} alt={title}></img>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
