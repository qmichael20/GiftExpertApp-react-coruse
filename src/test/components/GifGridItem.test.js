import { GifGridItem } from "../../components/GifGridItem";
import React from "react";
import { shallow } from "enzyme";

describe("Pruedas en GifGridItem", () => {
  const title = "Titulo";
  const url = "https://localhost/imagen.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Componente renderizado correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Url y alt deben ser igual a props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("animacion animate__pulse", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__pulse")).toBe(true);
  });
});
