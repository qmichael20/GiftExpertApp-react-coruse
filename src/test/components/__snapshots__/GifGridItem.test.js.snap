// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruedas en GifGridItem Componente renderizado correctamente 1`] = `
<div
  align="center"
  className="card animate__animated animate__pulse"
>
  <img
    alt="Titulo"
    src="https://localhost/imagen.jpg"
  />
</div>
`;
